File: langs/python3/problems/file_storage/stub.py
class Simulation:
    def __init__(self):
        pass

    def add_file(self, name, size):
        """Admin add. "true" if created, "false" if the name exists."""
        raise NotImplementedError

    def get_file_size(self, name):
        """Size as a string, or "" if missing."""
        raise NotImplementedError

    def delete_file(self, name):
        """Deleted size as a string, or "" if missing."""
        raise NotImplementedError

    def copy_file(self, source, dest):
        """Copy size to dest. Dest size, or "" if missing or over capacity."""
        raise NotImplementedError

    def get_n_largest(self, prefix, n):
        """Up to n files as name(size), ... Empty string if none.

        Example: "/dir/file2(20), /dir/deeper/file3.mov(9)".
        """
        raise NotImplementedError

    def add_user(self, user_id, capacity):
        """Create a user. "true", or "false" if the id exists."""
        raise NotImplementedError

    def add_file_by(self, user_id, name, size):
        """Store as that user. Remaining capacity, or ""."""
        raise NotImplementedError

    def merge_user(self, user_id1, user_id2):
        """Merge user_id2 into user_id1, then delete user_id2.

        Remaining capacity of user_id1, or "".
        """
        raise NotImplementedError

    def backup_user(self, user_id):
        """Snapshot that user's files. File count, or "" if missing."""
        raise NotImplementedError

    def restore_user(self, user_id):
        """Restore latest backup. Count, "0" if none, or "" if missing."""
        raise NotImplementedError

File: langs/java/problems/file_storage/stub.java
public class Simulation {
    public Simulation() {}

    /**
     * Admin add. Returns {@code "true"} if created, {@code "false"} if
     * the name exists.
     */
    public String addFile(String name, int size) {
        return "";
    }

    /**
     * Returns the size as a string, or {@code ""} if missing.
     */
    public String getFileSize(String name) {
        return "";
    }

    /**
     * Returns the deleted size as a string, or {@code ""} if missing.
     */
    public String deleteFile(String name) {
        return "";
    }

    /**
     * Copy size to dest. Returns dest size, or {@code ""} if source is
     * missing or the dest owner is over capacity.
     */
    public String copyFile(String source, String dest) {
        return "";
    }

    /**
     * Up to n files as {@code name(size), ...}. Empty string if none.
     * Example: {@code "/dir/file2(20), /dir/deeper/file3.mov(9)"}.
     */
    public String getNLargest(String prefix, int n) {
        return "";
    }

    /**
     * Create a user. Returns {@code "true"}, or {@code "false"} if the
     * id exists.
     */
    public String addUser(String userId, int capacity) {
        return "";
    }

    /**
     * Store a file as that user. Returns remaining capacity, or
     * {@code ""} if missing, name taken, or over capacity.
     */
    public String addFileBy(String userId, String name, int size) {
        return "";
    }

    /**
     * Merge {@code userId2} into {@code userId1}, then delete
     * {@code userId2}. Returns remaining capacity of {@code userId1},
     * or {@code ""}.
     */
    public String mergeUser(String userId1, String userId2) {
        return "";
    }

    /**
     * Snapshot that user's files. Returns the file count, or {@code ""}
     * if missing.
     */
    public String backupUser(String userId) {
        return "";
    }

    /**
     * Restore the latest backup. Returns the count, {@code "0"} if none,
     * or {@code ""} if missing.
     */
    public String restoreUser(String userId) {
        return "";
    }
}
